juju/juju
# provider/azure: update Azure/autorest SDKs

## Description of change

Update to the latest azure-sdk-for-go and go-autorest packages.
There are a lot of changes as the generated methods signatures
have changed significantly. This SDK change is necessary to get
access to the managed disks APIs, so we can support releasing
and importing volumes.

The one major functional change is that go-autorest now handles
HTTP status code 429 (TooManyRequests), so we can drop the
gross "callAPI" code from Juju. The tests we had for handling
rate-limiting have been dropped, since it's handled by go-autorest.

## QA steps

1. juju bootstrap azure azure1
2. juju add-model kubernetes
3. juju deploy canonical-kubernetes
4. juju deploy postgresql --storage pgdata=azure,10G
(wait for all to settle)
5. juju bootstrap azure azure2
6. juju switch azure1
7. juju migrate kubernetes azure2
(wait for migration to complete)
8. juju destroy-controller -y azure1
(check that migrated resources are still intact)
9. juju switch azure2:kubernetes
10. juju destroy-model -y kubernetes --destroy-storage
(wait for completion, confirm the model's resource group is gone)
11. juju destroy-controller -y azure2

## Documentation changes

None.

## Bug reference

None.
